apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/ProducerConstructorMapInterceptor.java

```diff
@@ -22,16 +22,27 @@
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
 import org.apache.skywalking.apm.util.StringUtil;
 
+import java.util.List;
 import java.util.Map;
+import java.util.regex.Pattern;
 
+/**
+ * Target at kafka-clients 2.1+
+ */
```

Comment:
Why add this comment? This interceptor is used for 2.1-, isn't it?

Reply:
> Why add this comment? This interceptor is used for 2.1-, isn't it?

No. The `Map` argument is used from Kafka 2.1+

Reply:
But this interceptor still supports `String` as a parameter, I still can see the String casting.

Reply:
You didn't add any new instrumentation definition. So, this must be compatible with older releases.

Reply:
> But this interceptor still supports `String` as a parameter, I still can see the String casting.

Exactly. Regarding `bootstrap.servers`, the type of the value in the map can be `String` or `List`. Both types are supported by kafka client lib.

Reply:
I am not talking about the codes. You added comment says this is for 2.1+, which is not accurate. It supports 2.1 also.

Reply:
> I am not talking about the codes. You added comment says this is for 2.1+, which is not accurate. It supports 2.1 also.

So it should be ">=2.1.0", I guess?

Reply:
Fixed